src/lt/objs/deploy.cljs

```diff
@@ -21,31 +21,64 @@
 
 (def shell (load/node-module "shelljs"))
 (def fs (js/require "fs"))
+; Update the comment for the following to describe the full situation:
+(def original-fs (js/require "original-fs")) ;; Required per [fs - reading asar files fails · Issue #1658 · atom/electron](https://github.com/atom/electron/issues/1658)
 (def zlib (js/require "zlib"))
 (def request (load/node-module "request"))
-(def tar (load/node-module "tar"))
+(def tar (load/node-module "tar-fs"))
+(def zip (load/node-module "node-stream-zip"))
 (def home-path (files/lt-home ""))
 ;; TODO: get-proxy
 ;; (def get-proxy (.-App.getProxyForURL (js/require "nw.gui")))
 (def get-proxy)
 
-(defn tar-path [v]
+; TODO: Cleanup platform name code
+
+; TODO: Rename to release-file-extension
+(def tar-extension (if (platform/win?) ".zip" ".tar.gz"))
+
+(defn get-release-download-url
+  "Retrieve the download URL for the specified release version and this computer's platform."
+  [v cb]
+  (let [release-info-url (str "https://api.github.com/repos/LightTable/LightTable/releases/tags/"
+                              v)]
+    (fetch/xhr release-info-url
+               {}
+               (fn [data]
```

Comment:
I do agree this suggestion is harder to read, but feels cleaner once you stare at it for a few minutes. In particular, using the results of `->` as the first argument to `->>` was initially disorientating. 

Maybe use a let binding instead of: 
```
(-> (js/JSON.parse data)
      (js->clj :keywordize-keys true))
```
to improve readability?

Not to self-promote poorly written blog posts (of which I am a fine provider of), but I wrote a rather brief guide about [[arrow macros once](http://bauerspace.com/target-practice-with-the-arrow-macro/).